src/components/tabs/js/tabsController.js

```diff
@@ -97,6 +99,92 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
     delete $attrs.$mdTabsTemplate;
   }
 
+  var DOMtabs = [];
+
+  function handleScroll () {
+    var nbTabs = ctrl.tabs.length;
+
+    var watcher = $scope.$watch(function () {
+      return $element.find('md-tab-content').length;
+    }, function (nv) {
+      if (nv === nbTabs) {
+        watcher();
+        refreshDOMtabs();
+      }
+    });
+  }
+
+  function refreshDOMtabs () {
+    DOMtabs = $element.find('md-tab-content');
+    if (ctrl.shrinkToolbar) {
+      bindScrollListener(ctrl.selectedIndex);
+    }
+  }
+
+  function bindScrollListener (newIndex, oldIndex) {
+    if (!DOMtabs.length) return;
+    if (oldIndex !== undefined && ctrl.tabs[oldIndex]) {
+      angular.element(DOMtabs[oldIndex]).off('scroll');
+      $timeout(function () {
+        scrollTop(oldIndex);
+      }, 1000);
+    }
+
+    prevScrollTop = 0;
+    angular.element(DOMtabs[newIndex || 0])
+      .off('scroll')
+      .on('scroll', shrinkToolbar);
+  }
+
+  var prevScrollTop = 0;
+  var shrinkSpeedFactor = .5;
+  var contentScroll = 0;
+  var toolbar;
+  var toolbarHeight = 0;
+  var y = 0;
+  var ZCLASS = 'md-whiteframe-z1';
+
+  function shrinkToolbar (e) {
+    if (!toolbar) {
+      elements.toolbar = angular.element($element.parent()[0].querySelector('md-toolbar'));
```

Comment:
@DevVersion the current "toolbar shrink" only listens for scroll on a parent md-content which does not apply when using tabs. The shrink behavior is actually very linked to the tabs themselves (keeping the scroll when changing tab, only listening to the current tab scroll etc.) But if you think it can be done in the toolbar component then feel free to use this PR as inspiration only :)